### PR TITLE
KeyError of Elemental references in pre-computed phase diagram

### PR DESCRIPTION
**Pull request by peikai**
## 

When I try to plot a pre-computed phase diagrams that are retrieved via get_phase_diagram_from_chemsys() method,

> phaseDiagram = mpr.get_phase_diagram_from_chemsys('Li-Fe-O')
> plotter = PDPlotter(phaseDiagram)

a KeyError raises as shown below, which arises from the inconsistence on object stored in el_refs dictionary, i.e., the keys stored in phaseDiagram.el_refs are **str** object (_dict_keys(['O', 'Fe', 'Li'])_), instead of the **Element** object (_dict_keys([Element O, Element Fe, Element Li])_).

The pre-computed phase diagrams in database might not be updated conveniently. So, I add some codes in pymatgen.analysis.phase_diagram.PhaseDiagram class, to update keys in elemental reference dictionary to Element object, in case it loads a pre-computed phase diagram. A corresponding test unit is also added to check the keys in el_refs dictionary,

KeyError message:

>Traceback (most recent call last):
  File "D:\Seafile\peikai\My Libraries\Repos\Volume-Planning-for-Anodes\phase_diagram_mixscheme.py", line 37, in <module>
    plotter = PDPlotter(phase_diagram, show_unstable=0, backend='matplotlib')
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Anaconda\envs\pymatgen2023\Lib\site-packages\pymatgen\analysis\phase_diagram.py", line 2111, in __init__
    self._min_energy = min(self._pd.get_form_energy_per_atom(e) for e in self._pd.stable_entries)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Anaconda\envs\pymatgen2023\Lib\site-packages\pymatgen\analysis\phase_diagram.py", line 2111, in <genexpr>
    self._min_energy = min(self._pd.get_form_energy_per_atom(e) for e in self._pd.stable_entries)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Anaconda\envs\pymatgen2023\Lib\site-packages\pymatgen\analysis\phase_diagram.py", line 577, in get_form_energy_per_atom
    return self.get_form_energy(entry) / entry.composition.num_atoms
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Anaconda\envs\pymatgen2023\Lib\site-packages\pymatgen\analysis\phase_diagram.py", line 564, in get_form_energy
    return entry.energy - sum(comp[el] * self.el_refs[el].energy_per_atom for el in comp.elements)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Anaconda\envs\pymatgen2023\Lib\site-packages\pymatgen\analysis\phase_diagram.py", line 564, in <genexpr>
    return entry.energy - sum(comp[el] * self.el_refs[el].energy_per_atom for el in comp.elements)
                                         ~~~~~~~~~~~~^^^^
KeyError: Element Li

### CODE REVIEW

**Comment by munrojm** [comment]
@peikai, I appreciate you making this PR. I just patched the API client to temporarily resolve this issue yesterday as we were having plotting issues on the front-end with the pre-computed data. There is an additional change that needs to be made to fix things for ternary compounds. `qhull_data` needs to be a numpy array, and not a list. To fix your local problems before this is merged, you can update `mp-api` as well.

@janosh, this relates to the emmet issue I tagged you in regarding `PhaseDiagram` de-serialization problems with monty.

**Comment by janosh** [comment]
@peikai Thanks for this PR. Do we really need `pymatgen/analysis/tests/pre_computed_phase_diagram_Li-Fe-O.json`? Maybe you can re-use an existing test file in the `test_files/` dir? The repo size is already approaching 1GB which we really need to get down.

File: pymatgen/analysis/tests/test_phase_diagram.py [hunk 1]
```diff
@@ -621,6 +623,14 @@ def test_read_json(self):
             dumpfn(self.pd, "pd.json")
             loadfn("pd.json")
 
+    def test_el_refs(self):
+        # load a pre_computed phase diagram,
+        # which was retrieved via new API and serialized to the MSONable object
+        with open(module_dir / "pre_computed_phase_diagram_Li-Fe-O.json") as pd:
+            phaseDiagram = json.load(pd, cls=MontyDecoder)
+        # check the keys in el_refs dict have been updated to Element object
+        assert str not in [type(el) for el in phaseDiagram.el_refs.keys()]
```
Comment (janosh) [inline pymatgen/analysis/tests/test_phase_diagram.py:632]:
```suggestion
        assert all(isinstance(el, Element) for el in phaseDiagram.el_refs)
```

**Review by janosh — CHANGES_REQUESTED** [review]

File: pymatgen/analysis/tests/test_phase_diagram.py (hunk 1 above)
Comment (peikai) [reply pymatgen/analysis/tests/test_phase_diagram.py]: I noticed the pre_computed_phase_diagram_Li-Fe-O.json is about 25Mb. I deleted it and utilized existed entries to make an imitation of pre_computed phase diagram with the KeyError issue. Thanks!

**Comment by peikai** [comment]
@munrojm Thanks for the information. The updated mp-api works. This PR can fix it in pymatgen.

**Review by janosh — APPROVED** [review]

**Pull request merged**